src/feditest/nodedrivers/imp/__init__.py

```diff
@@ -91,29 +85,21 @@ def perform_webfinger_query(
         if pair:
             ret_pair = HttpRequestResponsePair(first_request, current_request, pair.response)
             if ret_pair.response is not None:
-                if ret_pair.response.http_status == 200:
-                    if (
-                        not check_content_type
-                        or ret_pair.response.content_type() == "application/jrd+json"
-                        or ret_pair.response.content_type().startswith(
-                            "application/jrd+json;"
-                        )
-                    ):
-                        if ret_pair.response.payload is not None:
-                            json_string = ret_pair.response.payload.decode(
-                                encoding=ret_pair.response.payload_charset() or "utf8" )
-                            jrd = ClaimedJrd(json_string)
-                            if validate_jrd:
-                                try:
-                                    jrd.validate()
-                                except Exception as ex:
-                                    raise AssertionError(*ex.args[1:])
-                            return WebFingerQueryResponse(pair, jrd)
-                        raise WebFingerClient.WebfingerQueryFailedError(
-                            uri, ret_pair, "No payload"
-                        )
+                if ret_pair.response.http_status != 200:
+                    raise WebFingerClient.WebfingerQueryFailedError(uri, ret_pair, f"Invalid HTTP status: { ret_pair.response.http_status }")
+
+                if( ret_pair.response.content_type() != "application/jrd+json"
+                    and not ret_pair.response.content_type().startswith( "application/jrd+json;" )
+                ):
```

Comment:
This looks like a behavior change, but I'll run a local test to verify. The `check_content_type` option defaults to `False` and effectively changes that to `True`, which would cause some of the tests to report "shadowed' failures (failing for the wrong reason).